src/org/zaproxy/zap/extension/websocket/WebSocketAPI.java

```diff
@@ -0,0 +1,444 @@
+        }
+        return false;
+    }
+
+    private boolean sendWebSocketMessage(WebSocketProxy proxy, String message) throws IOException {
+        return this.sendWebSocketMessage(proxy, false, message);
+    }
+
+    private boolean sendWebSocketMessage(WebSocketProxy proxy, boolean outgoing, String message) throws IOException {
+        WebSocketMessageDTO resp = new WebSocketMessageDTO();
+        resp.channel = proxy.getDTO();
+        resp.payload = message;
+        resp.payloadLength = message.length();
+        resp.opcode = WebSocketMessage.OPCODE_TEXT;
+        resp.isOutgoing = outgoing;
+        boolean sent = proxy.send(resp, Initiator.WEB_SOCKET);
+        if (sent) {
+            try {
+                extension.recordMessage(resp);
+            } catch (DatabaseException e) {
+                LOG.error(e.getMessage(), e);
+            }
+        }
+        return sent;
+    }
+
+    private Map<Integer, EventConsumer> evMap = new HashMap<Integer, EventConsumer>();
+
+    private EventConsumer getEventConsumer(int channelId) {
+        EventConsumer ev = evMap.get(channelId);
+        if (ev == null) {
+            ev = new WebsocketEventConsumer(channelId);
+            evMap.put(channelId, ev);
+        }
+        return ev;
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        ApiResponse result;
+
+        if (VIEW_CHANNELS.equals(name)) {
+            ApiResponseList resultList = new ApiResponseList(name);
+            try {
+                List<WebSocketChannelDTO> channels = extension.getChannels(new WebSocketChannelDTO());
+                for (WebSocketChannelDTO channel : channels) {
+                    Map<String, String> map = new HashMap<String, String>();
+                    map.put("id", Integer.toString(channel.id));
+                    map.put("displayName", channel.toString());
+                    map.put("connected", Boolean.toString(channel.isConnected()));
+                    map.put("inScope", Boolean.toString(channel.isInScope()));
+                    map.put("contextUrl", channel.getContextUrl());
+                    map.put("fullUrl", channel.getFullUri());
+                    if (channel.getHandshakeReference() != null) {
+                        map.put("handshakeHistoryId", Integer.toString(channel.getHandshakeReference().getHistoryId()));
+                    }
+                    resultList.addItem(new ApiResponseSet<String>("channel", map));
+                }
+                result = resultList;
+            } catch (DatabaseException e) {
+                LOG.error(e.getMessage(), e);
+                throw new ApiException(ApiException.Type.INTERNAL_ERROR);
+            }
+        } else if (VIEW_MESSAGE.equals(name)) {
+            try {
+                WebSocketMessageDTO message = extension
+                        .getWebsocketMessage(params.getInt(PARAM_MESSAGE_ID), params.getInt(PARAM_CHANNEL_ID));

```

Comment:
`ApiUtils.getIntParam(params, PARAM_MESSAGE_ID)` (same for `PARAM_CHANNEL_ID`) for better handling of the int conversion.